pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+
+	"github.com/golang/glog"
+	sriovnetworkv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+)
+
+type MellanoxPlugin struct {
+	PluginName     string
+	SpecVersion    string
+	DesireState    *sriovnetworkv1.SriovNetworkNodeState
+	LastState      *sriovnetworkv1.SriovNetworkNodeState
+}
+
+var Plugin MellanoxPlugin
+
+// Initialize our plugin and set up initial values
+func init() {
+	Plugin = MellanoxPlugin{
+		PluginName:  "mellanox_plugin",
+		SpecVersion: "1.0",
+	}
+}
+
+// Name returns the name of the plugin
+func (p *MellanoxPlugin) Name() string {
+	return p.PluginName
+}
+
+// SpecVersion returns the version of the spec expected by the plugin
+func (p *MellanoxPlugin) Spec() string {
+	return p.SpecVersion
+}
+
+// OnNodeStateAdd Invoked when SriovNetworkNodeState CR is created, return if need dain and/or reboot node
+func (p *MellanoxPlugin) OnNodeStateAdd(state *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("mellanox-plugin OnNodeStateAdd()")
+
+	return p.OnNodeStateChange(nil, state)
+}
+
+// OnNodeStateChange Invoked when SriovNetworkNodeState CR is updated, return if need dain and/or reboot node
+func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("mellanoxi-Plugin OnNodeStateChange()")
+	needDrain = false
+	needReboot = false
+	err = nil
+	var mstconfigArgs []string
+
+	for _, iface := range new.Status.Interfaces {
+		if (iface.Vendor != "15b3") {
+			continue
+		}
+		for _, ifaceSpec := range new.Spec.Interfaces {
+			mstconfigArgs = []string{}
+			if (iface.PciAddress != ifaceSpec.PciAddress) {
+				continue
+			}
+			if (iface.TotalVfs != ifaceSpec.TotalVfs) {
+				mstconfigArgs = append(mstconfigArgs, fmt.Sprintf("NUM_OF_VFS=%s", ifaceSpec.TotalVfs))
+			}
+			if (iface.EnableSriov != ifaceSpec.EnableSriov) {
+				mstconfigArgs = append(mstconfigArgs, fmt.Sprintf("SRIOV_EN=%s", ifaceSpec.EnableSriov))
+			}
+			if (iface.LinkType != ifaceSpec.LinkType) {
+//				mstconfigArgs = append(mstconfig, "LINK_TYPE" + ifaceSpec.LinkType)
+			}
+			if (len(mstconfigArgs) > 0) {
```

Comment:
you should query mstconfig to get the values and set them according to NumVfs.
MAX_NUM_OF should be == NumVfs.
LinkType == Ethernet (fror now)
EnableSriov == Yes all the time.